native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -1063,19 +1041,158 @@ mod test {
 
     #[test]
     #[cfg_attr(miri, ignore)] // miri can't call foreign function `ZSTD_createCCtx`
-    #[cfg(not(target_os = "macos"))] // Github MacOS runner fails with "Too many open files".
```

Comment:
This test was previously failing because it created too many spill files. It now passes.